R/metareactive.R

```diff
@@ -81,7 +81,7 @@ metaReactive <- function(expr, env = parent.frame(), quoted = FALSE,
   #
   # Even though expr itself is quoted, wrapExpr will effectively unquote it by
   # interpolating it into the `metaExpr()` call, thus quoted = FALSE.
-  expr <- wrapExpr(shinymeta::metaExpr, expr, env, quoted = FALSE, localize = localize, bindToReturn = bindToReturn)
+  expr <- wrapExpr(shinymeta::metaExpr, expr, quoted = FALSE, localize = localize, bindToReturn = bindToReturn)
```

Comment:
Why do we no longer pass `env` along to `metaExpr()`?

Reply:
The `env` in this case, is supposed to be the parent environment for the `metaReactive`. (Remember, `metaReactive`, like `reactive`, does introduce a variable scope.) When we passed `env` into `wrapExpr`, then this code:

```r
env1 <- new.env()
metaReactive({ ... }, env = env1)
```

effectively became this:

```r
metaReactive2({
  metaExpr({ ... }, env = env1)
}, env = env1)
```

But that's actually NOT what we want. Inside the `metaReactive2`, `environment()` is not `env1`, but a new child environment of `env1`. We want `metaExpr` to execute in that child environment (which it does by default once we stop passing `env` explicitly to `metaExpr`).

I've added all sorts of unit tests, so you can see the failures that occur if you were to undo my change on this line.

Note that this issue has always existed, but was cancelled out by the fact that `metaExpr` didn't execute in the `env` you passed it, but rather in a temporary child environment; this is the `eval_tidy` vs. `eval` thing. But once I fixed that issue, then this incorrect behavior became evident.